src/functions/public/Export-Json.ps1

```diff
@@ -0,0 +1,160 @@
+function Export-Json {
+    <#
+        .SYNOPSIS
+        Exports JSON data to a file.
+
+        .DESCRIPTION
+        Converts PowerShell objects to JSON format and writes them to one or more files.
+        Supports various formatting options including indentation types, sizes, and compact output.
+        Can accept both PowerShell objects and JSON strings as input.
+
+        .EXAMPLE
+        Export-Json -InputObject $myObject -Path 'output.json'
+
+        Exports a PowerShell object to output.json with default formatting.
+
+        .EXAMPLE
+        Export-Json -InputObject $data -Path 'config.json' -IndentationType Spaces -IndentationSize 2
+
+        Exports data to config.json with 2-space indentation.
+
+        .EXAMPLE
+        Export-Json -JsonString $jsonText -Path 'data.json' -Compact
+
+        Exports a JSON string to data.json in compact format.
+
+        .EXAMPLE
+        $objects | Export-Json -Path 'output-{0}.json'
+
+        Exports multiple objects to numbered files via pipeline.
+
+        .EXAMPLE
+        Export-Json -InputObject $config -Path 'settings.json' -IndentationType Tabs -Force
+
+        Exports configuration to settings.json with tab indentation, overwriting if it exists.
+
+        .LINK
+        https://psmodule.io/Json/Functions/Export-Json/
+    #>
+
+    [CmdletBinding(DefaultParameterSetName = 'FromObject', SupportsShouldProcess)]
+    param (
+        # PowerShell object to convert and export as JSON.
+        [Parameter(Mandatory, ValueFromPipeline, ParameterSetName = 'FromObject')]
+        [PSObject]$InputObject,
+
+        # JSON string to export to file.
+        [Parameter(Mandatory, ValueFromPipeline, ParameterSetName = 'FromString')]
+        [string]$JsonString,
+
+        # The path to the output JSON file. Supports placeholders for pipeline processing.
+        [Parameter(Mandatory)]
+        [string]$Path,
+
+        # Produce compact (minified) output.
+        [Parameter()]
+        [switch]$Compact,
+
+        # Indentation type: 'Spaces' or 'Tabs'.
+        [Parameter()]
+        [ValidateSet('Spaces', 'Tabs')]
+        [string]$IndentationType = 'Spaces',
+
+        # Number of spaces or tabs per indentation level. Only used if not compacting.
+        [Parameter()]
+        [UInt16]$IndentationSize = 4,
+
+        # The maximum depth to serialize nested objects.
+        [Parameter()]
+        [int]$Depth = 100,
+
+        # Overwrite existing files without prompting.
+        [Parameter()]
+        [switch]$Force,
+
+        # Text encoding for the output file.
+        [Parameter()]
+        [ValidateSet('UTF8', 'UTF8BOM', 'UTF8NoBOM', 'UTF32', 'Unicode', 'ASCII')]
```

Comment:
Lets support the following encoding types (as supported by the Set-Content function.

ASCII, BigEndianUnicode, BigEndianUTF32, OEM, Unicode, UTF7, UTF8, UTF8BOM, UTF8NoBOM, UTF32

Default should be utf8NoBOM